src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -75,6 +77,13 @@ public void onPost() {
      * Called when a POST is made.
      */
     public void onPost(String triggeredByUser) {
+        onPost(SCMEvent.originOf(Stapler.getCurrentRequest()), triggeredByUser);
+    }
+
+    /**
+     * Called when a POST is made.
+     */
+    public void onPost(final String origin, String triggeredByUser) {
```

Comment:
S maybe replace it with more common object with fields that will be expandable in future?